_posts/2024-10-24-ruby-on-guard-rails.md

```diff
@@ -0,0 +1,281 @@
+
+- 👮‍♀️ [**`rubocop`**](https://rubocop.org): the best Ruby linter.
+  I generally try to enable as much as possible in Rubocop and disable rules locally when necessary.
+- 🪴 [**`erb_lint`**](https://github.com/Shopify/erb_lint): like Rubocop, but for ERB.
+  Helps keep your view templates a bit more consistent.
+- 💐 [**`better_html`**](https://github.com/Shopify/better-html): helps keep your HTML a bit more consistent through development-time checks.
+- 🖖 [**`prosopite`**](https://github.com/charkost/prosopite): avoids N+1 queries in development and test environments.
+- 🪪 [**`licensed`**](https://github.com/github/licensed): ensures that all of your dependencies are licensed correctly.
+- 🤖 [**`actionlint`**](https://github.com/rhysd/actionlint): ensures that your GitHub Actions workflows are correct.
+- 🇪🇸 [**`eslint`**](https://eslint.org): when you inevitably have to write some JavaScript: lint that too.
+
+I add these linters to my Gemfile with something like this:
+
+```ruby
+group :development do
+  gem "better_html"
+  gem "erb_lint"
+  gem "licensed"
+  gem "rubocop-capybara"
+  gem "rubocop-performance"
+  gem "rubocop-rails"
+  gem "rubocop-rspec"
+  gem "rubocop-rspec_rails"
+end
+```
+
+If you want to enable/disable more Rubocop rules, remember to do something like this:
+
+```yaml
+require:
+  - rubocop-performance
+  - rubocop-rails
+  - rubocop-rspec
+  - rubocop-rspec_rails
+  - rubocop-capybara
+
+AllCops:
+  TargetRubyVersion: 3.3
+  ActiveSupportExtensionsEnabled: true
+  NewCops: enable
+  EnabledByDefault: true
+
+Layout:
+  Exclude:
+    - "db/migrate/*.rb"
+```
+
+Note, this will almost certainly enable things you don't want.
+That's fine, disable them manually.
+Here you can see we've disabled all `Layout` cops on database migrations (as they are generated by Rails).
+
+Consider disabling lints on a per-file basis when possible:
+
+```ruby
+# Bulk create BrewCommandRuns for each Device.
+# Since there are no callbacks or validations on
+# BrewCommandRun, we can safely use insert_all!
+#
+# rubocop:disable Rails/SkipsModelValidations
+BrewCommandRun.insert_all!(new_brew_command_runs)
+# rubocop:enable Rails/SkipsModelValidations
+```
+
+I always recommend a comment explaining why you're disabling the linter in this particular case.
+
+### 🧪 tests
+
+I define "tests" as anything that requires the developer to actually write additional, non-production code to catch problems.
+In my opinion, you want as few of these as you can to maximally exercise your codebase.
+
+- 🧪 [**`rspec`**](https://rspec.info): the Ruby testing framework used by most Ruby projects I've worked on.
+  Minitest is fine, too.
+- 🙈 [**`simplecov`**](https://github.com/colszowka/simplecov): the standard Ruby code coverage tool.
+  Integrates with other tools (like CodeCov) and allows you to enforce code coverage.
+- 🕴️ [**`playwright`**](https://playwright.dev): dramatically better than Selenium for Rails system tests with JavaScript.
```

Comment:
Another emoji suggestion! The Playwright logo are two masks not unsimilar to this emoji

```suggestion
- 🎭 [**`playwright`**](https://playwright.dev): dramatically better than Selenium for Rails system tests with JavaScript.
```